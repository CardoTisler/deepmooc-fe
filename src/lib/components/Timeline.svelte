<script lang="ts">
	import type { TimelineEventData } from '$lib/types';
	import TimelineEvent from '$lib/components/TimelineEvent.svelte';
	import Section from './util/Section.svelte';
	import theme from '$lib/shared/stores/theme';

	$: isContrastMode = $theme === 'contrast';
	export let eventsList: TimelineEventData[];
</script>

<Section class="flex flex-col ml-12 p-4 max-w-lg {isContrastMode ? 'black-bg' : ''}">
	<div>
		<p class="text-lg font-thin dark:text-yellow-300">AJAJOON</p>
	</div>
	{#each eventsList as data}
		<TimelineEvent {data} />
	{/each}
</Section>
