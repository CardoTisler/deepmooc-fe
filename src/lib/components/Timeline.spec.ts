import type { TimelineEventData } from '$lib/types';
import { render } from '@testing-library/svelte';
import Timeline from './Timeline.svelte';

describe('Timeline.svelte', () => {
	it('should render component', () => {
		const mockEventsList: TimelineEventData[] = [
			{
				title: 'TEST TITLE',
				subTitle: 'TEST SUBTITLE',
				deadlineTime: '00:00:00',
				deadlineDate: '01.01.1970',
				timeRemaining: '10h',
				tags: ['TEST TAG']
			}
		];

		const { container } = render(Timeline, { eventsList: mockEventsList });
		expect(container).toMatchSnapshot();
	});
});
