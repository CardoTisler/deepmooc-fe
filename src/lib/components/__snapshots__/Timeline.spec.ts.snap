// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Timeline.svelte > should render component 1`] = `
<body>
  <div>
    <div
      class="border-2 border-gray-200 rounded-xl bg-white w-full dark:border-yellow-300 flex flex-col ml-12 p-4 max-w-lg  s-u_aBuH8kPm0b"
    >
      <div>
        <p
          class="text-lg font-thin dark:text-yellow-300"
        >
          AJAJOON
        </p>
      </div>
       
      <div
        class="flex flex-col p-2 dark:text-yellow-300"
      >
        <div
          class="flex flex-row"
        >
          <div
            class="flex flex-row pl-3 w-full justify-between"
          >
            <div>
              <h3>
                TEST TITLE
              </h3>
            </div>
             
            <div>
              <a
                aria-label="lahenda ülesannet"
                href="/courses/169/assignments/69"
              >
                <svg
                  clip-rule="evenodd"
                  fill="black"
                  fill-rule="evenodd"
                  height="24"
                  stroke="black"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.883 12l-7.527 6.235.644.765 9-7.521-9-7.479-.645.764 7.529 6.236h-21.884v1h21.883z"
                  />
                </svg>
                <!--&lt;RightArrow&gt;-->
              </a>
            </div>
          </div>
        </div>
         
        <div
          class="flex flex-col border-l border-gray-400 pl-8"
        >
          <div>
            <p
              class="gray-spaced dark:text-yellow-300"
            >
              TEST SUBTITLE
            </p>
          </div>
           
          <div
            class="flex flex-row"
          >
            <div
              class="rounded-full bg-gray-200 px-2 dark:bg-black mr-2 h-fit"
            >
              <span
                class="text-gray-600 dark:text-yellow-300"
              >
                TEST TAG
              </span>
            </div>
            <!--&lt;Pill&gt;-->
          </div>
           
          <div
            class="grid grid-cols-2 pt-4"
          >
            <div
              class="pb-2"
            >
              <p
                class="gray-spaced dark:text-yellow-300"
              >
                TÄHTAEG
              </p>
            </div>
             
            <div>
              <p
                class="gray-spaced dark:text-yellow-300"
              >
                AEGA JÄÄNUD
              </p>
            </div>
             
            <div>
              <p
                class="font-medium"
              >
                00:00:00
              </p>
            </div>
             
            <div>
              <p
                class="font-medium"
              >
                10h
              </p>
            </div>
             
            <div>
              01.01.1970
            </div>
          </div>
        </div>
      </div>
      <!--&lt;TimelineEvent&gt;-->
      
    </div>
    <!--&lt;Section&gt;-->
    <!--&lt;Timeline&gt;-->
  </div>
</body>
`;

exports[`Timeline.svelte > should render component with contrast color scheme 1`] = `
<body>
  <div>
    <div
      class="border-2 border-gray-200 rounded-xl bg-white w-full dark:border-yellow-300 flex flex-col ml-12 p-4 max-w-lg black-bg s-u_aBuH8kPm0b"
    >
      <div>
        <p
          class="text-lg font-thin dark:text-yellow-300"
        >
          AJAJOON
        </p>
      </div>
       
      <div
        class="flex flex-col p-2 dark:text-yellow-300"
      >
        <div
          class="flex flex-row"
        >
          <div
            class="flex flex-row pl-3 w-full justify-between"
          >
            <div>
              <h3>
                TEST TITLE
              </h3>
            </div>
             
            <div>
              <a
                aria-label="lahenda ülesannet"
                href="/courses/169/assignments/69"
              >
                <svg
                  clip-rule="evenodd"
                  fill="yellow"
                  fill-rule="evenodd"
                  height="24"
                  stroke="yellow"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.883 12l-7.527 6.235.644.765 9-7.521-9-7.479-.645.764 7.529 6.236h-21.884v1h21.883z"
                  />
                </svg>
                <!--&lt;RightArrow&gt;-->
              </a>
            </div>
          </div>
        </div>
         
        <div
          class="flex flex-col border-l border-gray-400 pl-8"
        >
          <div>
            <p
              class="gray-spaced dark:text-yellow-300"
            >
              TEST SUBTITLE
            </p>
          </div>
           
          <div
            class="flex flex-row"
          >
            <div
              class="rounded-full bg-gray-200 px-2 dark:bg-black mr-2 h-fit"
            >
              <span
                class="text-gray-600 dark:text-yellow-300"
              >
                TEST TAG
              </span>
            </div>
            <!--&lt;Pill&gt;-->
          </div>
           
          <div
            class="grid grid-cols-2 pt-4"
          >
            <div
              class="pb-2"
            >
              <p
                class="gray-spaced dark:text-yellow-300"
              >
                TÄHTAEG
              </p>
            </div>
             
            <div>
              <p
                class="gray-spaced dark:text-yellow-300"
              >
                AEGA JÄÄNUD
              </p>
            </div>
             
            <div>
              <p
                class="font-medium"
              >
                00:00:00
              </p>
            </div>
             
            <div>
              <p
                class="font-medium"
              >
                10h
              </p>
            </div>
             
            <div>
              01.01.1970
            </div>
          </div>
        </div>
      </div>
      <!--&lt;TimelineEvent&gt;-->
      
    </div>
    <!--&lt;Section&gt;-->
    <!--&lt;Timeline&gt;-->
  </div>
</body>
`;
